File: Easy/LC1614/LC1614.java
class Solution {
    public int maxDepth(String s) {
        if(s.length() <= 1) return 0;
        int currDepth = 0;
        int maxDepth = 0;
        for(int i=0; i<s.length(); i++) {
            if(s.charAt(i) == '(') {
                currDepth += 1;
                maxDepth = Math.max(maxDepth, currDepth);
            } else if (s.charAt(i) == ')') {
                currDepth--;
            }
        }
        return maxDepth;
    }
}

File: Easy/LC1614/LC1614.js
/**
 * @param {string} s
 * @return {number}
 */
const maxDepth = s => {
    let maxDepth = 0;
    s.split('').reduce((acc, curr) => {
        if(curr === '(') {
            acc++;
            maxDepth = Math.max(maxDepth, acc);
        } else if(curr === ')') acc--;
        return acc;
    }, 0);
    return maxDepth;
};

const maxDepth = s => {
    let maxDepth = 0;
    let currDepth = 0;
    for(let c of s) {
        if(c === '(') {
            currDepth++;
            maxDepth = Math.max(maxDepth, currDepth);
        } else if(c === ')') currDepth--;
    }
    return maxDepth;
};